shared_model/interfaces/queries/query.hpp

```diff
@@ -65,17 +66,34 @@ namespace shared_model {
        * system queries plus 1. Required for preventing replay attacks.
        * @return attached query counter
        */
-      virtual const QueryCounterType &queryCounter() = 0;
+      virtual const QueryCounterType &queryCounter() const = 0;
 
       // ------------------------| Primitive override |-------------------------
 
       std::string toString() const override {
-        return boost::apply_visitor(detail::ToStringVisitor(), get());
+        return detail::PrettyStringBuilder()
+            .init("Query")
+            .append("creator_id", creatorAccountId())
+            .append("query_counter", std::to_string(this->queryCounter()))
+            .append(Signable::toString())
+            .append("domain_data",
+                    boost::apply_visitor(detail::ToStringVisitor(), get()))
+            .finalize();
       }
 
       OldModelType *makeOldModel() const {
-        return boost::apply_visitor(
+         auto old_model = boost::apply_visitor(
             detail::OldModelCreatorVisitor<OldModelType *>(), get());
+        old_model->creator_account_id = creatorAccountId();
+        old_model->query_counter = queryCounter();
+        // signature related
+        old_model->created_ts = createdTime();
+        std::for_each(signatures().begin(), signatures().end(), [&old_model](auto &signature_wrapper) {
+          auto old_sig = signature_wrapper->makeOldModel();
+          old_model->signature = *old_sig;
```

Comment:
Old model has only one signature so this behavior will assign a last signature from a new model. I think we should decide, should we change query to be like an old model query and have only a single signature or at least make this assignment explicit.